odl/discr/lp_discr.py

```diff
@@ -74,37 +77,50 @@ def __init__(self, fspace, grid, dspace, exponent=2.0, interp='nearest',
             'nearest' : use nearest-neighbor interpolation (default)
 
             'linear' : use linear interpolation
+
         order : {'C', 'F'}, optional
-            Ordering of the values in the flat data arrays. 'C'
-            means the first grid axis varies slowest, the last fastest,
-            'F' vice versa.
+            Ordering of the axes in the data storage. 'C' means the
+            first axis varies slowest, the last axis fastest;
+            vice versa for 'F'.
         """
         if not isinstance(fspace, FunctionSpace):
-            raise TypeError('{!r} is not a `FunctionSpace` instance.'
+            raise TypeError('{!r} is not a FunctionSpace instance.'
                             ''.format(fspace))
         if not isinstance(fspace.domain, IntervalProd):
             raise TypeError('Function space domain {!r} is not an '
-                            '`IntervalProd` instance.'.format(fspace.domain))
+                            'IntervalProd instance.'.format(fspace.domain))
+        if not isinstance(partition, RectPartition):
+            raise TypeError('Partition {!r} is not a RectPartition '
+                            'instance.'.format(partition))
+        if not fspace.domain.contains_set(partition.bbox):
+            raise ValueError('Partition {} is not a subset of the function '
```

Comment:
Well currently it's equivalent but I usually only used !r when raising a `TypeError`, otherwise just pretty-printing. Shall we discuss a general guideline?